dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java

```diff
@@ -323,6 +324,8 @@ private void executeComplementProcess() throws Exception {
             }
             // flow end
             // execute next process instance complement data
+            ProcessDefinition processDefinition = processService.findProcessDefineById(processDefinitionId);
```

Comment:
Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

Reply:
> Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

I also have the same question, ^-^

Reply:
@dailidong @JinyLeeChina  @caoping6 
It seems a same problem described in #5909
I personally think that the problem comes in  [MasterExecThread.java#L977](https://github.com/apache/dolphinscheduler/blob/1887bde1ebf8249de153890e78d449582e3eaf4a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java#L977)
aftert this:
`processInstance = processService.findProcessInstanceById(processInstance.getId());`
the `processDefinition` variable in `ProcessInstance` woule be null.
```java
// ProcessInstance.java
    /**
     * process definition structure
     */
    @TableField(exist = false)
    private ProcessDefinition processDefinition;
```